framework/src/main/java/org/checkerframework/framework/ajava/JointJavacJavaParserVisitor.java

```diff
@@ -0,0 +1,2382 @@
+        if (javacTree.getElseStatement() != null) {
+            javacTree.getElseStatement().accept(this, node.getElseStmt().get());
+        }
+
+        return null;
+    }
+
+    @Override
+    public Void visitImport(ImportTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof ImportDeclaration)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, ImportDeclaration.class);
+        }
+
+        ImportDeclaration node = (ImportDeclaration) javaParserNode;
+        processImport(javacTree, node);
+        // In javac trees, a name like a.* is stored as a member select, but JavaParser just stores
+        // a and records that the name ends in an asterisk.
+        if (node.isAsterisk()) {
+            assert javacTree.getQualifiedIdentifier().getKind() == Kind.MEMBER_SELECT;
+            MemberSelectTree identifier = (MemberSelectTree) javacTree.getQualifiedIdentifier();
+            identifier.getExpression().accept(this, node.getName());
+        } else {
+            javacTree.getQualifiedIdentifier().accept(this, node.getName());
+        }
+
+        return null;
+    }
+
+    @Override
+    public Void visitInstanceOf(InstanceOfTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof InstanceOfExpr)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, InstanceOfExpr.class);
+        }
+
+        InstanceOfExpr node = (InstanceOfExpr) javaParserNode;
+        processInstanceOf(javacTree, node);
+        javacTree.getExpression().accept(this, node.getExpression());
+        javacTree.getType().accept(this, node.getType());
+        return null;
+    }
+
+    @Override
+    public Void visitIntersectionType(IntersectionTypeTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof IntersectionType)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, IntersectionType.class);
+        }
+
+        IntersectionType node = (IntersectionType) javaParserNode;
+        processIntersectionType(javacTree, node);
+        visitLists(javacTree.getBounds(), node.getElements());
+        return null;
+    }
+
+    @Override
+    public Void visitLabeledStatement(LabeledStatementTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof LabeledStmt)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, LabeledStmt.class);
+        }
+
+        LabeledStmt node = (LabeledStmt) javaParserNode;
+        processLabeledStatement(javacTree, node);
+        javacTree.getStatement().accept(this, node.getStatement());
+        return null;
+    }
+
+    @Override
+    public Void visitLambdaExpression(LambdaExpressionTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof LambdaExpr)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, LambdaExpr.class);
+        }
+
+        LambdaExpr node = (LambdaExpr) javaParserNode;
```

Comment:
Minor:  I would find it easier to read the code if variables like `node` were prefixed by `javaParser` or `javac`.  In this case, a clearer name would be `javaParserLambdaExpr`.  (Do you prefer the generic `node` so all method bodies use the same variable name for the downcasted variable?)

Reply:
Yes, I prefer that name because it can be used in every method, and the name `javaParserNode` is already used for the parameter type.